src/MSBuild/XMake.cs

```diff
@@ -2366,6 +2366,11 @@ string commandLine
                         out enableProfiler
                         );
 
+                    if (Traits.Instance.DebugEngine)
+                    {
+                        verbosity = LoggerVerbosity.Diagnostic;
```

Comment:
If this doesn't mess up the verbosity settings for other attached loggers, it needs a comment explaining why please.